plugins/services/src/js/components/forms/GeneralServiceFormSection.js

```diff
@@ -227,6 +233,15 @@ class GeneralServiceFormSection extends Component {
         <p>Constraints control where apps run to allow optimization for either fault tolerance or locality.</p>
         {this.getPlacementConstraintsFields(data.constraints)}
         <FormRow>
+          <FormGroup
+            className="column-12"
+            showError={constraintsErrors != null && !Array.isArray(constraintsErrors)}>
+            <FieldError>
+              {constraintsErrors}
+            </FieldError>
+          </FormGroup>
+        </FormRow>
```

Comment:
Sorry, I know we did this together but I just realized that if there is no error, then this empty row causes some additional margin to stack up. Could we conditionally render this entire row, dependent on the existence of the error?